app/javascript/hooks/queries/shared_accesses/useShareableUsers.jsx

```diff
@@ -1,16 +1,17 @@
 import { useQuery } from 'react-query';
 import axios from '../../../helpers/Axios';
 
-export default function useShareableUsers(friendlyId, input) {
+export default function useShareableUsers(friendlyId, input, isThreeCharacters) {
   const params = {
     search: input,
   };
 
   return useQuery(
-    ['getShareableUsers', input],
+    ['getShareableUsers', { ...params }],
     () => axios.get(`/shared_accesses/${friendlyId}/shareable_users.json`, { params }).then((resp) => resp.data.data),
     {
       keepPreviousData: true,
+      enabled: isThreeCharacters,
```

Comment:
`isThreeCharacters` seems to be an information on the `input` data,
Therefore `useShareableUsers` can extract it without the need to pass in an extra data?

Reply:
It's also used in the view.

Reply:
You would rather duplicate the logic inside the hook?

Reply:
I think I'd do that,
First the logic here isn't expensive nor complex,
The `isThreeCharacters` itself will be provided in accordance to the `input`,
Whenever the latter changes the other changes,
The complexity of adding that extra variable in memory may exceed what you're trying to gain in terms of performance,

Reply:
I'm sorry - I don't understand what you mean.

Is adding an extra param, an extra variable in memory that big of a deal?

I'm not a fan of duplicating the logic of the view inside the hook.

Reply:
I think I see what you mean. Because isThreeChar is dependant on input. And both are given to the hook as params. Yeah ok.

I still think adding the variable makes senses and makes it more enjoyable to read/understand. I didn't add it for performance reasons.

Reply:
What you're suggesting is good and I'd go that way myself,
However, the notes are caused by the naming schema.
By indicating `isThreeCharacters` we've already gave it a sense and that made a inner dependency for the given arguments (`isThreeCharachters` depends on `input`) and that what led me to spot this in the first place.
However, if you'd simply call that `enabled` instead of `isThreeCharachters` on the hook signature - that would brake the dependency logic.
It's true that `enabled` will map to `isThreeCharachters` in value  but only in this context which is irrelevant to the hook, It's semantically different,
If we're calling it as `isThreeCharachters` in the hook then it's an information on the `input` that we can extract with no complexity,
To further understand this, think about the implication of both schemas when reusing the hook in another context.
One can map the `enabled` to any value to enable/disable the mutation and another will be tight to the what `isThreeCharachers` means.

Reply:
Yeah agreed. I would just check the input length directly in the mutation. No need to pass it in